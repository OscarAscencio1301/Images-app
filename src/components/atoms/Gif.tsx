

const Gif = ({title, url}:{ id?: string; title: string; url: string; }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

export default Gif
