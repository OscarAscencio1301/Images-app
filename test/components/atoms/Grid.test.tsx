import { render, screen } from '@testing-library/react'
import Gif from '../../../src/components/atoms/Gif'

describe("Test Atom Grid", () => {
    const title = 'Nuevo Gif';
    const url = 'https://www.adslzone.net/app/uploads-adslzone.net/2019/04/borrar-fondo-imagen.jpg';

    test('Snapshot to component', () => {
        const { container } = render(<Gif title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })
    test('Debe de mostrar la imagen con el url indicado y el alt', () => {
        render(<Gif title={title} url={url} />)
        // expect(screen.getByRole<HTMLImageElement>('img').src).toBe(url)
        // expect(screen.getByRole<HTMLImageElement>('img').alt).toBe(title)

        const {src, alt} = screen.getByRole<HTMLImageElement>('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })

    test('Debe de mostrar el tÃ­tulo en el componente', () => {
        render(<Gif title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy()
    })
    


})