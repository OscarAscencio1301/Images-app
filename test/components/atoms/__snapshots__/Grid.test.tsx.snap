// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Atom Grid Snapshot to component 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Nuevo Gif"
      src="https://www.adslzone.net/app/uploads-adslzone.net/2019/04/borrar-fondo-imagen.jpg"
    />
    <p>
      Nuevo Gif
    </p>
  </div>
</div>
`;
